(DOCSP-11049) Added note on macOS permission for Compass to run

jocelyn-mendez opened the pull request:
Added note that macOS Catalina, and later versions, may require permission to be granted to allow Compass to run. 

jira: https://jira.mongodb.org/browse/DOCSP-11049

staging: https://docs-mongodbcom-staging.corp.mongodb.com/compass/jocelynmendez/DOCSP-11049/index.html

jeff-allen-mongo commented on source/upgrade.txt:
```diff
@@ -99,6 +99,14 @@ Procedure
 
+.. note::
+
+   If you are running macOS Catalina, or later versions, you may have to modify
```
[nits] 

- I think we have too many commas here which makes the pace of reading this a little off.

- I don't think we need to say "later versions" we can just say "or later".

Suggest something like:

`If you are running macOS Catalina or later, you may...`

jeff-allen-mongo commented on source/upgrade.txt:
```diff
@@ -99,6 +99,14 @@ Procedure
 
 #. Launch the fully-featured edition of |compass|.
 
+.. note::
```
So, as is we have this note tucked in the "migration" section of this page. This note doesn't really apply to the migration procedure. Instead, I'd suggest moving this to the bottom of the `Manually Install the Latest Version of Compass` section, since if they are manually installing that's when they are most likely to run into this issue.

jocelyn-mendez commented:
Hey @jeff-allen-mongo, thanks for the feedback! I have gone through and made the changes, so please feel free to let me know your thoughts when you get a chance.

Updated staging: https://docs-mongodbcom-staging.corp.mongodb.com/compass/jocelynmendez/DOCSP-11049/upgrade.html

jocelyn-mendez commented:
Hey @mmarcon, I have added the noted regarding permission to run Compass on macOS Catalina or later. Please let me know your thoughts when you get a chance, Thanks!

Updated staging: https://docs-mongodbcom-staging.corp.mongodb.com/compass/jocelynmendez/DOCSP-11049/upgrade.html

Jira: https://jira.mongodb.org/browse/DOCSP-11049
